tests/flava/test_modeling_flava.py

```diff
@@ -0,0 +1,1234 @@
+                        self.text_model_tester.vocab_size,
+                    ),
+                )
+        else:
+            self.parent.assertIsNone(result.text_embeddings)
+
+        if test_image and test_text:
+            self.parent.assertEqual(
+                result.multimodal_masked_embeddings.shape,
+                (
+                    self.multimodal_model_tester.batch_size,
+                    self.text_model_tester.seq_length + num_patches + 2,
+                    self.multimodal_model_tester.hidden_size,
+                ),
+            )
+            self.parent.assertEqual(
+                result.itm_logits.shape,
+                (self.text_model_tester.batch_size, 2),
+            )
+            self.parent.assertEqual(
+                result.mmm_text_logits.shape,
+                (
+                    (inputs["mlm_labels"] != self.multimodal_model_tester.ce_ignore_index).sum().item(),
+                    self.text_model_tester.vocab_size,
+                ),
+            )
+            self.parent.assertEqual(
+                result.mmm_image_logits.shape,
+                (inputs["bool_masked_pos"].sum().item(), self.image_model_tester.vocab_size),
+            )
+            self.parent.assertEqual(
+                result.contrastive_logits_per_image.shape,
+                (self.image_model_tester.batch_size, self.text_model_tester.batch_size),
+            )
+            self.parent.assertEqual(
+                result.contrastive_logits_per_text.shape,
+                (self.text_model_tester.batch_size, self.image_model_tester.batch_size),
+            )
+
+            for item in [
+                result.loss_info.global_contrastive,
+                result.loss_info.itm,
+                result.loss_info.mmm_text,
+                result.loss_info.mmm_image,
+            ]:
+                self.parent.assertEqual(item.dim(), 0)
+
+            for item in [result.loss_info.mim, result.loss_info.mlm]:
+                self.parent.assertIsNone(item)
+
+        else:
+            self.parent.assertIsNone(result.multimodal_masked_embeddings)
+            for item in [
+                result.loss_info.global_contrastive,
+                result.loss_info.itm,
+                result.loss_info.mmm_text,
+                result.loss_info.mmm_image,
+            ]:
+                self.parent.assertIsNone(item)
+
+        self.parent.assertIsNone(result.multimodal_embeddings)
+
+
+@require_torch
+class FLAVAForPreTrainingTest(FLAVAModelTest):
+    all_model_classes = (FLAVAForPreTraining,) if is_torch_available() else ()
+    class_for_tester = FLAVAForPreTrainingTester
+    test_torchscript = False
+
+
+# We will verify our results on an image of cute cats
+def prepare_img():
+    url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+    im = Image.open(requests.get(url, stream=True).raw)
+    return im
+
+
+@require_vision
+@require_torch
+class FLAVAModelIntegrationTest(unittest.TestCase):
```

Comment:
Shouldn't Flava also work for Text classification (GLUE) or image classification? Can't be add those heads here as well?

Reply:
And the corresponding classes?

Reply:
They should! They are planned for a separate PR. I am trying to unblock some use cases by first adding the pretraining classes.

Reply:
Ok that's fine by me!